### PR TITLE
Promote cri-api to v1beta1

### PR DESCRIPTION
**Pull request by mrunalp**
Signed-off-by: Mrunal Patel <mpatel@redhat.com>
/sig node

@saschagrunert @mikebrow 

**What type of PR is this?**

/kind api-change

**What this PR does / why we need it**:
This moves the CRI API to `v1beta1` for https://github.com/kubernetes/enhancements/issues/2040

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Promote cri-api from v1alpha2 to v1beta1
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

- [KEP]: https://github.com/kubernetes/enhancements/issues/2040

```docs
Promote cri-api from v1alpha2 to v1beta1
```

### CODE REVIEW

File: pkg/kubelet/cm/cpumanager/cpu_manager.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/klog/v2"
 
-	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1beta1"
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/cm/cpumanager/cpu_manager.go:30]: would be handy to add this to `hack/.import-aliases`

**Comment by SergeyKanzhelev** [comment]
Compared with this PR: https://github.com/kubernetes/kubernetes/pull/58973/ and everything looks ok. The only place still mentions api-cri/v1alpha1 is here `pkg/kubelet/kuberuntime/kuberuntime_container_windows.go`

> https://github.com/kubernetes/kubernetes/blob/56d1c3b96d0a544130a82caad33dd57629b8a7f8/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto#L681-L682

But since the sha of merged commit is not known, I guess it needs to wait to be updated.

/retest

**Comment by derekwaynecarr** [comment]
i think we need to ship both protos during a transition period.

**Comment by SergeyKanzhelev** [comment]
> i think we need to ship both protos during a transition period.

for my education, should the previous protos be obtained by referencing older version of [cri-api](https://github.com/kubernetes/cri-api)? kubelet will use a single API for making calls, right?

**Comment by mrunalp** [comment]
@SergeyKanzhelev @derekwaynecarr I opened https://github.com/kubernetes/kubernetes/pull/96387 to add `v1beta1` alongside `v1alpha2`. We can hold this PR for the eventual changes to switch to v1beta1.

**Comment by SergeyKanzhelev** [comment]
> > i think we need to ship both protos during a transition period.
> 
> for my education, should the previous protos be obtained by referencing older version of [cri-api](https://github.com/kubernetes/cri-api)? kubelet will use a single API for making calls, right?

>> @SergeyKanzhelev @derekwaynecarr I opened #96387 to add `v1beta1` alongside `v1alpha2`. We can hold this PR for the eventual changes to switch to v1beta1.

So the idea is to keep using `v1alpa2` while switching runtimes to the beta? This is why both are needed in-tree?

**Comment by mrunalp** [comment]
@SergeyKanzhelev The tests are failing because containerd in CI doesn't yet implement `v1beta1` yet. So, the idea is that we add `v1beta` proto, update containerd and CRI-O to support `v1beta1` and then switch the kubelet to use `v1beta1` and drop `v1alpha2`. Meanwhile, runtimes could support both versions in parallel and then drop `v1alpha2` support later release. We don't run into this with docker because dockershim is in tree and this PR update dockershim as well to use `v1beta1` avoiding this transition.

```
Nov 09 18:14:07 kind-control-plane kubelet[499]: E1109 18:14:07.733484     499 remote_runtime.go:83] Version from runtime service failed: rpc error: code = Unimplemented desc = unknown service runtime.v1beta1.RuntimeService
Nov 09 18:14:07 kind-control-plane kubelet[499]: E1109 18:14:07.734093     499 kuberuntime_manager.go:200] Get runtime version failed: get remote runtime typed version failed: rpc error: code = Unimplemented desc = unknown service runtime.v1beta1.RuntimeService
Nov 09 18:14:07 kind-control-plane kubelet[499]: F1109 18:14:07.734150     499 server.go:270] failed to run Kubelet: failed to create kubelet: get remote runtime typed version failed: rpc error: code = Unimplemented desc = unknown service runtime.v1beta1.RuntimeService
```

**Comment by SergeyKanzhelev** [comment]
> @SergeyKanzhelev The tests are failing because containerd in CI doesn't yet implement `v1beta1` yet. So, the idea is that we add `v1beta1` proto, update containerd and CRI-O to support `v1beta1` in CI and then switch the kubelet to use `v1beta1` and drop `v1alpha2`. Meanwhile, runtimes could support both versions in parallel and then drop `v1alpha2` support in a later release. We don't run into this with docker because dockershim is in tree and this PR updates dockershim as well to use `v1beta1` avoiding needing this transition.
> 
> ```
> Nov 09 18:14:07 kind-control-plane kubelet[499]: E1109 18:14:07.733484     499 remote_runtime.go:83] Version from runtime service failed: rpc error: code = Unimplemented desc = unknown service runtime.v1beta1.RuntimeService
> Nov 09 18:14:07 kind-control-plane kubelet[499]: E1109 18:14:07.734093     499 kuberuntime_manager.go:200] Get runtime version failed: get remote runtime typed version failed: rpc error: code = Unimplemented desc = unknown service runtime.v1beta1.RuntimeService
> Nov 09 18:14:07 kind-control-plane kubelet[499]: F1109 18:14:07.734150     499 server.go:270] failed to run Kubelet: failed to create kubelet: get remote runtime typed version failed: rpc error: code = Unimplemented desc = unknown service runtime.v1beta1.RuntimeService
> ```

Good exercise coordinating version updates =). I wonder how it was possible before here: https://github.com/kubernetes/kubernetes/pull/58973 Perhaps there were no ContainerD tests.

**Comment by mrunalp** [comment]
> Good exercise coordinating version updates =). I wonder how it was possible before here: #58973 Perhaps there were no ContainerD tests.

Yeah, there were no containerd tests blocking back then  :)

**Comment by derekwaynecarr** [comment]
Just recording discussion from SIG Node on 11/10:

- recommendation is to move to v1 package in 1.20
- runtimes add server side support for v1 package during 1.21 phase
- kubelet switches client to v1 packages during 1.21 after cri-o/containerd has updated CI
- kubelet advertises cri as beta in 1.21

Outcome:
- kubelet client speaks one version (latest)
- runtimes may speak multiple versions (community choice)

**Comment by SergeyKanzhelev** [comment]
/close

since the v1 was introduced instead https://github.com/kubernetes/kubernetes/pull/96387/, this is not needed

**Pull request closed without merging**